src/main/webapp/script.js

```diff
@@ -31,6 +31,89 @@ function displayLatitudeLongitude(value) {
   document.getElementById('longitude').value = value['lng'];
 }
 
+// Update displayed COVID stats based on coordinates
+function displayLocationData(value) {
+  const potentialReports = [];
+  const RADIUS = 1.5;
+  // Look for closest reports
+  fetch('/report').then((response) => response.json()).then((reports) => {
+    reports.forEach((report) => {
+      if (Math.abs(report.lat - value['lat']) < RADIUS &&
+          Math.abs(report.lng - value['lng']) < RADIUS) {
+        potentialReports.push(report);
+      }
+    });
+
+    // If there are no nearby reports, dispay global data
+    if (potentialReports.length == 0) {
+      displayCurrentStats(
+          'Worldwide', globalActive, globalConfirmed, globalDeaths,
+          globalRecovered);
+      return;
+    }
+    // If there are nearby reports, lookup address using geocoder
+    const lookupURL =
+        'https://maps.googleapis.com/maps/api/geocode/json?latlng=' +
+        value['lat'] + ',' + value['lng'] + '&key=' +
+        mykey.substring(44, 83);  // Indices represent the actual API key
+    let potentialReport = potentialReports[0];
+    let foundFlag = false;
+    // Compare location names with territory names from potential reports
+    fetch(lookupURL).then((response) => response.json()).then((data) => {
+      data.results[0].address_components.forEach((location) => {
+        potentialReports.forEach((report) => {
+          // If there is a match, display that territory's statistics
+          if (location.long_name.trim().valueOf().includes(
```

Comment:
if you're gonna use the value twice (location.long_name.trim().valueOf() and same for territory), consider storing them as a variable, that way you're only running the chain of methods once and its not as cluttered